opentelemetry/proto/metrics/v1/metrics.proto

```diff
@@ -118,255 +145,122 @@ message MetricDescriptor {
-    // values, it calculates configurable percentiles over a sliding time
-    // window.
-    // Corresponding values are stored in SummaryDataPoint.
-    SUMMARY = 7;
+    // INSTANTANEOUS is a metric whose values are measured at a particular
+    // instant.
+    //
+    // If an INSTANTANEOUS metric is reported over any time interval the
+    // values are representative of distinct measurements within the
+    // interval.
+    INSTANTANEOUS = 1;
+
+    // DELTA is a metric whose values are measured and successively added
+    // over a reporting interval. The reported value represents the change
+    // from the previous interval.
+    //
+    // Each reported value for a DELTA metric over a time interval is
+    // self-contained, the values do not depend on values in other time
+    // intervals. This is unlike a CUMMULATIVE metric where values are sums
+    // of all measured events in the time interval and all previously
+    // measured values of the metric.
+    //
+    // DELTA metrics are not guaranteed to be monotonic.
+    DELTA = 2;
+
+    // CUMULATIVE is a metric whose values are measured and successively
+    // added to the lifetime total of the metric.
+    //
+    // Each reported value for a CUMULATIVE metric is the sum of all
+    // measurements up to and including that one.
+    //
+    // CUMULATIVE metrics are not guaranteed to be monotonic.
+    CUMULATIVE = 3;
   }
-  Type type = 4;
+  Interval interval = 4;
 
-  // The set of labels associated with the metric descriptor. Labels in this list apply to
-  // all data points.
-  repeated opentelemetry.proto.common.v1.StringKeyValue labels = 5;
-}
+  // monotonic is true if the metric values are entirely non-increasing, or
+  // entirely non-decreasing.
+  bool monotonic = 5;
 
-// Int64DataPoint is a single data point in a timeseries that describes the time-varying
-// values of a int64 metric.
-message Int64DataPoint {
-  // The set of labels that uniquely identify this timeseries.
-  repeated opentelemetry.proto.common.v1.StringKeyValue labels = 1;
-
-  // start_time_unix_nano is the time when the cumulative value was reset to zero.
-  // This is used for Counter type only. For Gauge the value is not specified and
-  // defaults to 0.
-  //
-  // The cumulative value is over the time interval (start_time_unix_nano, time_unix_nano].
-  // Value is UNIX Epoch time in nanoseconds since 00:00:00 UTC on 1 January 1970.
-  //
-  // Value of 0 indicates that the timestamp is unspecified. In that case the timestamp
-  // may be decided by the backend.
-  fixed64 start_time_unix_nano = 2;
-
-  // time_unix_nano is the moment when this value was recorded.
-  // Value is UNIX Epoch time in nanoseconds since 00:00:00 UTC on 1 January 1970.
-  fixed64 time_unix_nano = 3;
-
-  // value itself.
-  int64 value = 4;
-}
+  // Subset describes the subset of numbers metric values belong to if any.
+  enum Subset {
```

Comment:
What about `NumericQuality` to go along with the proposed `TemporalQuality` ?

Reply:
@c24t : When I looked into implementing the `refinement` approach I wasn't sure how to restrict  mutually exclusive refinements using the flag system. Using the enums like the current approach builds sets restrictions using syntax, but I'm not sure how to achieve the same otherwise (i.e. you could set both `monotonic_increasing` and `monotonic_decreasing`).

Do you have a way in mind to prevent this with this kind of implementation? Want to make sure I didn't miss something (because I like the uniformity of this approach).

Reply:
AFAIK there's no way to to prevent invalid bitsets in the proto itself, I assume it'd be up to the collector to return some sensible error.

One benefit of using flags instead of enums is that we can add refinements without updating the proto package. (Whether this is a benefit or a detriment maybe depends on your use case: it would be very bad news if the client and the collector interpreted this field differently!) But this is only true if the proto doesn't know anything about refinement semantics.

I don't have a strong opinion on this vs. enums, just pointing out that there's a tradeoff here between safety and development speed.